File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
      int w=0, l=0, r=height.size()-1;
        while(l<r){
            int a;
            // we  first have to check which element is < b/c the lesser element decides the height
            if(height[l] < height[r]){//if the element pointed by the l is < we pick it as height and increament the l ptr
            a= height[l] * (r-l);
                l++;
            }
            else{//if the element pointed by the r is < we pick it as height and decreament the r ptr
                a=height[r] * (r-l);
                r--;
            }
            w=max(w,a);//store max area
        }
        return w;
    }
};

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        start = 0 
        end = len(height) - 1
        maxi = 0
        while start < end:
            if height[start] < height[end]:
                width = end - start
                area = height[start] * width
                start += 1
            else:
                width = end - start
                area = height[end] * width
                end -= 1
            maxi = max(maxi, area)
                
        return maxi